packages/core-kit/src/nodes/fetch.ts

```diff
@@ -163,8 +217,28 @@ export default {
         ),
       };
     } else {
-      const response = raw ? await data.text() : await data.json();
-      return { response, status };
+      let response;
+      if (raw || !contentType || !contentType.includes("application/json")) {
+        response = await data.text();
+        return {
+          response,
+          status,
+          statusText,
+          contentType,
+          responseHeaders,
+        };
+      } else {
+        response = await data.json();
+        const responseSchema = generateJsonSchemaFromObject(response);
```

Comment:
not sure about this one

Reply:
which bit?

Reply:
The generating JSON schema of the response bit. Not sure why we would need that.

Reply:
So two use-cases I have right now
1. To feed into the object and array manipulation nodes for picking apart a response
2. For checking and then inputting the schema when dealing with a new response

Reply:
Got it. My spidey sense tells me that this is probably a different kind of node. Maybe a util node that takes a JSON object and produces a schema for it? Let's not add this bit to the fetch node.

Reply:
brilliant idea